File: Java/797. All Paths From Source to Target.java
class Solution {
    public List<List<Integer>> allPathsSourceTarget(int[][] graph) {
        List<List<Integer>> res = new ArrayList<>();
        List<Integer> temp = new ArrayList<>();
        int vis=graph.length-1;
        temp.add(0);
        solve(graph,0,res,temp,vis);
        return res;
    }
    public void solve(int[][] graph,int n,List<List<Integer>> res,List<Integer> temp,int vis){
        for(int j:graph[n]){
            temp.add(j);
            if(j==vis){
                res.add(new ArrayList<>(temp));
            }      
            solve(graph,j,res,temp,vis);        
            temp.remove(temp.size()-1); 
        }
    }
}

File: C++/797. All Paths From Source to Target.cpp
class Solution {
public:
    vector<vector<int>> allPathsSourceTarget(vector<vector<int>>& graph) {
        vector<int>temp;
        vector<vector<int>> res;
        int vis=graph.size()-1;
        solve(graph,0,res,temp,vis);
        return res;
    }
    void solve(vector<vector<int>>& graph,int n,vector<vector<int>> &res,vector<int>&temp,int vis){
        temp.push_back(n);
        for(int j:graph[n]){
            if(j==vis){
                temp.push_back(vis);
                res.push_back(temp);
                temp.pop_back();
                continue;
            }      
            solve(graph,j,res,temp,vis);        
            temp.pop_back();
        }
    }
};
